src/compiler/checker.ts

```diff
@@ -5047,8 +5014,25 @@ namespace ts {
         }
 
         function getTypeOfFuncClassEnumModule(symbol: Symbol): Type {
-            const links = getSymbolLinks(symbol);
+            let links = getSymbolLinks(symbol);
             if (!links.type) {
+                const jsDeclaration = getDeclarationOfJSInitializer(symbol.valueDeclaration);
+                if (jsDeclaration) {
+                    const jsSymbol = getMergedSymbol(jsDeclaration.symbol);
+                    if (jsSymbol && (jsSymbol.exports && jsSymbol.exports.size || jsSymbol.members && jsSymbol.members.size)) {

```

Comment:
If `jsSymbol === symbol` (somehow) does this work need to be done?

Reply:
No, and it would be really bad because you would be merging symbols with themselves. But it would hit the new assert in mergeSymbol, so it doesn’t happen in our tests at least.

But you make a valid point; getDeclarationOfJSInitializer should (1) check that its node is in a js file (2) check that the provided value declaration is the initializer of the js declaration that it returns.

Reply:
Fixed. (As is the call to getSymbolOfNode.)